File: bench/native/floatloop.c
// bench: 10M-iteration f64 while loop — C reference (self-timed).
#include <stdio.h>
#include <time.h>

int main(void) {
    clock_t t0 = clock();
    volatile double total = 0.0;
    double i = 0.0;
    while (i < 10000000.0) {
        total = total + i;
        i = i + 1.0;
    }
    double ms = (double)(clock() - t0) * 1000.0 / CLOCKS_PER_SEC;
    printf("%.0f\n", (double)total);
    fprintf(stderr, "time: %.3f ms\n", ms);
    return 0;
}

File: bench/native/floatloop.rs
// bench: 10M-iteration f64 while loop — Rust reference (self-timed).
fn main() {
    let t0 = std::time::Instant::now();
    let mut total: f64 = 0.0;
    let mut i: f64 = 0.0;
    while i < 10_000_000.0 {
        total = std::hint::black_box(total + i); // match the C version's volatile accumulator
        i += 1.0;
    }
    let ms = t0.elapsed().as_secs_f64() * 1000.0;
    println!("{:.0}", total);
    eprintln!("time: {:.3} ms", ms);
}
